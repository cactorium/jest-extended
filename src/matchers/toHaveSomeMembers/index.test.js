import matcher from './';

expect.extend(matcher);

describe('.toHaveSomeMembers', () => {
  const shallow = { hello: 'world' };
  const deep = { message: shallow };

  it('passes when given array is a subset', () => {
    expect([2, 3]).toHaveSomeMembers([1, 2, 3]);
  });

  it('passes when given array contains some of the members', () => {
    expect([1, 2, 3]).toHaveSomeMembers([2, 3, 4]);
  });

  it('passes when given array contains object value', () => {
    expect([shallow]).toHaveSomeMembers([shallow, deep]);
  });

  it('fails when given array does not contain any members ', () => {
    expect(() => expect([4, 5, 6]).toHaveSomeMembers([1, 2, 3])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given object is not an array', () => {
    expect(() => expect(7).toHaveSomeMembers([1, 2, 3])).toThrowErrorMatchingSnapshot();
  });

  it('fails when expected object does not contain array value', () => {
    expect(() => expect([1, 2, 3]).toHaveSomeMembers('world')).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toHaveSomeMembers', () => {
  const shallow = { hello: 'world' };

  it('passes when given array does not contain primitive value', () => {
    expect(['hola', 'mundo']).not.toHaveSomeMembers(['hello', 'world']);
  });

  it('passes when given array does not contain object value', () => {
    expect([{ hello: 'world' }]).not.toHaveSomeMembers([{ world: 'hello' }]);
  });

  it('passes when given object is not array', () => {
    expect(7).not.toHaveSomeMembers([7]);
  });

  it('fails when given object contains primitive value', () => {
    expect(() => expect([7]).not.toHaveSomeMembers([7, 8])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given array contains object value', () => {
    expect(() => expect([{ foo: 'bar' }]).not.toHaveSomeMembers([{ foo: 'bar' }])).toThrowErrorMatchingSnapshot();
  });

  it('fails when given array contains array value', () => {
    expect(() => expect([[shallow]]).not.toHaveSomeMembers([[shallow], 7])).toThrowErrorMatchingSnapshot();
  });
});
