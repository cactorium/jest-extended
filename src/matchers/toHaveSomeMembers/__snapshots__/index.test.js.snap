// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveSomeMembers fails when given array contains array value 1`] = `
"<dim>expect(</><red>received</><dim>).not.toHaveSomeMembers(</><green>expected</><dim>)</>

Expected list to not have any members of:
  <green>[[{\\"hello\\": \\"world\\"}], 7]</>
Received:
  <red>[[{\\"hello\\": \\"world\\"}]]</>"
`;

exports[`.not.toHaveSomeMembers fails when given array contains object value 1`] = `
"<dim>expect(</><red>received</><dim>).not.toHaveSomeMembers(</><green>expected</><dim>)</>

Expected list to not have any members of:
  <green>[{\\"foo\\": \\"bar\\"}]</>
Received:
  <red>[{\\"foo\\": \\"bar\\"}]</>"
`;

exports[`.not.toHaveSomeMembers fails when given object contains primitive value 1`] = `
"<dim>expect(</><red>received</><dim>).not.toHaveSomeMembers(</><green>expected</><dim>)</>

Expected list to not have any members of:
  <green>[7, 8]</>
Received:
  <red>[7]</>"
`;

exports[`.toHaveSomeMembers fails when expected object does not contain array value 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveSomeMembers(</><green>expected</><dim>)</>

Expected object to have some member of :
  <green>\\"world\\"</>
Received:
  <red>[1, 2, 3]</>"
`;

exports[`.toHaveSomeMembers fails when given array does not contain any members  1`] = `
"<dim>expect(</><red>received</><dim>).toHaveSomeMembers(</><green>expected</><dim>)</>

Expected object to have some member of :
  <green>[1, 2, 3]</>
Received:
  <red>[4, 5, 6]</>"
`;

exports[`.toHaveSomeMembers fails when given object is not an array 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveSomeMembers(</><green>expected</><dim>)</>

Expected object to have some member of :
  <green>[1, 2, 3]</>
Received:
  <red>7</>"
`;
